mffpy/reader.py

```diff
@@ -236,3 +236,54 @@ def get_physical_samples_from_epoch(self, epoch: xml_files.Epoch,
         dt = dt if dt is None or 0.0 < dt < epoch.dt-t0 else None
         return self.get_physical_samples(
             t0, dt, channels, block_slice=epoch.block_slice)
+
+    def get_mff_content(self):
+        """return the content of an mff file.
+
+        The output of this function is supposed to return a dictionary
+        containing one serializable object per valid .xml file. Valid
+        .xml files are those whose types belongs to one of the available
+        XMLType sub-classes.
+
+        **Returns**
+        mff_content: dict
+            dictionary containing the content of an mff file.
+        """
+
+        # Root tags corresponding to available XMLType sub-classes
+        xml_root_tags = ['fileInfo', 'dataInfo', 'patient', 'sensorLayout',
+                         'coordinates', 'epochs', 'eventTrack', 'categories',
+                         'dipoleSet']
+        # Create the dictionary that will be returned by this function
+        mff_content = {tag: {} for tag in xml_root_tags}
+
+        # Iterate over existing .xml files
+        for xmlfile in self.directory.files_by_type['.xml']:
+            with self.directory.filepointer(xmlfile) as fp:
+                try:
+                    obj = XML.from_file(fp)
+                    content = obj.get_serializable_content()
+
+                    if obj.xml_root_tag == 'categories':
+                        # Add EEG data to each segment of each category
+                        for cat_obj in content['categories'].values():
+                            # Iterate over each segment
+                            for i in range(len(cat_obj)):
+                                # Get samples from epoch
+                                samples = self.get_physical_samples_from_epoch(
+                                    self.epochs[i], channels=['EEG'])
+                                eeg, start_time = samples['EEG']
+                                # Insert an EEG data field into each segment
+                                cat_obj[i]['eegData'] = eeg.tolist()
```

Comment:
Any reason this isn't in `Categories.get_content` or `Categories.get_serializable_content`?